pkg/artifactory/datasource_artifactory_file.go

```diff
@@ -104,50 +105,52 @@ func dataSourceArtifactoryFile() *schema.Resource {
 	}
 }
 
-func dataSourceFileRead(d *schema.ResourceData, m interface{}) error {
+func dataSourceFileReader(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	repository := d.Get("repository").(string)
 	path := d.Get("path").(string)
 	outputPath := d.Get("output_path").(string)
 	forceOverwrite := d.Get("force_overwrite").(bool)
 	fileInfo := FileInfo{}
 	_, err := m.(*resty.Client).R().SetResult(&fileInfo).Get(fmt.Sprintf("artifactory/api/storage/%s/%s", repository, path))
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	fileExists := FileExists(outputPath)
 	chksMatches, _ := VerifySha256Checksum(outputPath, fileInfo.Checksums.Sha256)
 
-	if fileExists {
-		if !chksMatches && !forceOverwrite {
-			return fmt.Errorf("local file differs from upstream version and no overwrite is permitted")
-		}
-	} else {
+	/*--File Download logic--
+	1. File doesn't exist
+	2. In Data Source argument `force_overwrite` set to true, an existing file in the output_path will be overwritten. Ignore file exists or not
+	3. File exists but check sum doesn't match
+	*/
+	if !fileExists || forceOverwrite || (fileExists && !chksMatches) {
 		outdir := filepath.Dir(outputPath)
 		err = os.MkdirAll(outdir, os.ModePerm)
 		if err != nil {
-			return err
+			return diag.FromErr(err)
 		}
 		outFile, err := os.Create(outputPath)
 		if err != nil {
-			return err
+			return diag.FromErr(err)
 		}
-
 		defer func(outFile *os.File) {
 			_ = outFile.Close()
 		}(outFile)
+	} else { //download not required
+		return diag.FromErr(fmt.Errorf("err001: file download skiped. fileExists: %v, chksMatches: %v, forceOverwrite: %v", fileExists, chksMatches, forceOverwrite))
```

Comment:
Yeah, you can't do this - it isn't an error. 
It's informational - I think, but I am not sure, that you can return informational - But I definitely know you can return warning

Reply:
changed to `diag.Warning`